src/libraries/Native/Unix/CMakeLists.txt

```diff
@@ -10,7 +10,7 @@ elseif(CLR_CMAKE_TARGET_MACCATALYST)
 endif()
 cmake_policy(SET CMP0042 NEW)
 
-project(CoreFX C)
```

Comment:
I don't think we should remove the "C". All corefx sources are intentionally C code (because of Mono)

Reply:
In such case we need to figure out a different way to encode the HAVE_SIGACTION_ULONG_FLAGS check, I haven't found any way to do that with plain C due to its less stringent type conversion checks.

Reply:
Hmm, I don't understand then why the C compiler complains when compiling the actual code, but doesn't fail on the same thing in the check. I am starting to wonder whether the changes related to HAVE_SIGACTION_ULONG_FLAGS originated from the time when the libraries were still being compiled as C++ code. @smx-smx could it be the case?

Reply:
Hmm i can have the test work in plain C

```
$ arm-buildroot-linux-uclibcgnueabi-gcc -Werror=incompatible-pointer-types t.c -o t
$ gcc -Werror=incompatible-pointer-types t.c -o t
t.c: In function ‘main’:
t.c:5:32: error: initialization of ‘long unsigned int *’ from incompatible pointer type ‘int *’ [-Werror=incompatible-pointer-types]
    5 |         unsigned long *flags = &action.sa_flags;
      |                                ^
cc1: some warnings being treated as errors
```

Reply:
Ah, so maybe setting 
```
CMAKE_REQUIRED_FLAGS=-Werror=incompatible-pointer-types
```
before the test code and clearing it after could do the trick.